src/care/algorithm_impl.h

```diff
@@ -979,6 +979,20 @@ CARE_INLINE void fill_n(care::host_device_ptr<T> arr, Size n, const U& val)
    } CARE_STREAM_LOOP_END
 }
 
+/************************************************************************
+ * Function  : copy_n
+ * Author(s) : Alan Dayton
+ * Purpose   : Copies one ManagedArray into another.
+ * ************************************************************************/
+template <class T, class Size, class U>
+CARE_INLINE void copy_n(care::host_device_ptr<T> in, Size n,
+                        care::host_device_ptr<U> out)
+{
+   CARE_STREAM_LOOP(i, 0, n) {
+      out[i] = in[i];
+   } CARE_STREAM_LOOP_END
```

Comment:
Since this is inline, we don't need instantiations, right?

Reply:
I believe that is the case.